### PR TITLE
Release Candidate v2.2.2

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Release Candidate for v2.2.2

## Type of change

- New Features
   - netCDF4 files produced using `to_netcdf4()` now have an unlimited
     time dimension
- Documentation
   - Updated guidance on numpy version for installation
- Bug Fix
   - Updated madrigal methods to simplify compound data types and enable
     creation of netCDF4 files using `self.to_netcdf4()`.
   - Updated ICON support for latest file organization on SSL servers

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. Please also list any relevant details for
your test configuration

- nosetests and travisCI

**Test Configuration**:
* macOS 10.15
* python 2.7.18

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zeondo.json
- [x] doc tests pass locally
- [x] Upload the test distribution to the testPyPi server
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (include list of these packages)
- [x] Create a wiki page with the release text [https://github.com/pysat/pysat/wiki/v2.2.2-Release-Text]()
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [2.2.2] - 2020-12-31
```
Comment (rstoneback) [inline CHANGELOG.md:5]:
```suggestion
## [2.2.2] - 2020-11-23
```

File: .zenodo.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      }, 
+      {
+        "name": "vvonbose"
+      }, 
+      {
+        "affiliation": "Predictive Science", 
+        "name": "Asher Pembroke"
+      }, 
+      {
+        "name": "Gayatri"
```
Comment (rstoneback) [inline .zenodo.json:45]:
```suggestion
        "name": "Gayatri Iyer"
```

File: .zenodo.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      }, 
+      {
+        "name": "Asher Pembroke"
+      }, 
+      {
+        "affiliation": "University of Texas at Dallas", 
+        "name": "Matthew Depew"
+      }, 
+      {
+        "name": "NHargrave"
```
Comment (rstoneback) [inline .zenodo.json:30]:
```suggestion
        "name": "Nathaniel Hargrave"
```

File: .zenodo.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        "name": "Angeline Burrell",
+        "orcid": "0000-0001-8875-9326"
+      }, 
+      {
+        "name": "Carey Spence",
+        "orcid": "0000-0001-8340-5625"
+      }, 
+      {
+        "name": "Asher Pembroke"
+      },
```
Comment (rstoneback) [inline .zenodo.json:24]:
```suggestion
 
```
Asher in there twice.

File: .zenodo.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      {
+        "name": "NHargrave"
+      }, 
+      {
+        "affiliation": "Catholic University of America, Goddard Space Flight Center",
+        "name": "Jonathon Smith",
+        "orcid": "0000-0002-8191-4765"
+      }, 
+      {
+        "name": "vvonbose"
```
Comment (rstoneback) [inline .zenodo.json:38]:
```suggestion
        "name": "Veronica Von Bose"
```

File: .zenodo.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
+{
+    "creators": [
+      {
+        "affiliation": "University of Texas at Dallas",
```
Comment (rstoneback) [inline .zenodo.json:4]:
```suggestion
        "affiliation": "The University of Texas at Dallas", 
```

**Comment by rstoneback** [comment]
The docs build on my system and on RTD. I'm not sure what the `doc tests` are otherwise.

**Comment by rstoneback** [comment]
There are errors:
```
======================================================================
ERROR: Checking download routine functionality for module:  icon ivm  a
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/russellstoneback/opt/miniconda3/envs/standard2/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/russellstoneback/Code/pysat/pysat/tests/test_instruments.py", line 279, in check_download
    raise e
error_perm: 550 Failed to change directory.
-------------------- >> begin captured logging << --------------------
pysat: WARNING: Unable to find any files that match the supplied template. If you have the necessary files please check pysat settings and file locations (e.g. pysat.pysat_dir).
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: Checking download routine functionality for module:  icon ivm  b
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/russellstoneback/opt/miniconda3/envs/standard2/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/russellstoneback/Code/pysat/pysat/tests/test_instruments.py", line 279, in check_download
    raise e
error_perm: 550 Failed to change directory.
-------------------- >> begin captured logging << --------------------
pysat: WARNING: Unable to find any files that match the supplied template. If you have the necessary files please check pysat settings and file locations (e.g. pysat.pysat_dir).
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: Test combine_kp when all input is provided
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/russellstoneback/opt/miniconda3/envs/standard2/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/russellstoneback/Code/pysat/pysat/tests/test_sw.py", line 317, in test_combine_kp_all
    assert (kp_inst['Kp'] != self.combine['fill_val']).all()
AssertionError

----------------------------------------------------------------------
Ran 1828 tests in 1712.729s
```

**Comment by jklenzing** [comment]
The IVM files are failing because these are now zipped on the server.  We could port over the code from pysatNASA that fixes this, but the other files are going to change shortly as well.  I'd rather get a better fix in place for the others.

The kp issues look like those we documented in pysat/pysatSpaceWeather#10.  I think this was fixed in pysat/pysatSpaceWeather#15.

**Comment by jklenzing** [comment]
Happy to hear it works.  Currently working on updates to the ICON instruments over there.  I still don't know what was causing the kp error, but it seems to have gone away.  It could be related to something changing in numpy or similar.

**Comment by rstoneback** [comment]
> Happy to hear it works. Currently working on updates to the ICON instruments over there. 

Thanks for the heads up.

> I still don't know what was causing the kp error, but it seems to have gone away. It could be related to something changing in numpy or similar.

Hasn't that Kp error been popping in and out for awhile now? Is it intermittent server availability?

**Comment by aburrell** [comment]
From what I can tell, it's because different dates are tested and pysat used to/somethings still (???) assumes that all data between two dates is present.  I fixed this recently by turning on the update_files flag when loading the instrument.

**Comment by aburrell** [comment]
> The docs build on my system and on RTD. I'm not sure what the doc tests are otherwise.

Check out the warning messages on Travis.  It's for broken links, bad docstring formats, examples that don't work, etc.  I think most of them are broken links, which is why the test is allowed to fail at the moment.

**Comment by jklenzing** [comment]
> > The docs build on my system and on RTD. I'm not sure what the doc tests are otherwise.
> 
> 
> 
> Check out the warning messages on Travis.  It's for broken links, bad docstring formats, examples that don't work, etc.  I think most of them are broken links, which is why the test is allowed to fail at the moment.

The broken links are fixed in #602. Still in draft mode. Hoping to finish this today.

**Comment by rstoneback** [comment]
Hey Folks (@aburrell and @jklenzing). The test release of pysat is up at: https://test.pypi.org/project/pysat/2.2.10/

Remember the version issues last time? We can't use v2.2.2 on test pypi.

**Comment by rstoneback** [comment]
> Hey Folks (@aburrell and @jklenzing). The test release of pysat is up at: https://test.pypi.org/project/pysat/2.2.10/
> 
> Remember the version issues last time? We can't use v2.2.2 on test pypi.

Found an issue with ICON. Latest at: https://test.pypi.org/project/pysat/2.2.11rc1/

**Review by aburrell — COMMENTED** [review]
The code looks fine for v2.X.  I tested with ocbpy and it passes.  Are there other modules that need to be considered?

**Comment by rstoneback** [comment]
> The code looks fine for v2.X. I tested with ocbpy and it passes. Are there other modules that need to be considered?

I just finished up a successful run with the ICON IVM software using the test pypi. Starting a run with cosmic.

**Comment by rstoneback** [comment]
COSMIC-2 run worked on my end.

**Review by aburrell — APPROVED** [review]

**Review by jklenzing — APPROVED** [review]

**Comment by rstoneback** [comment]
Thanks @aburrell and @jklenzing!

**Pull request merged**